Move theming support from Beta to Main

jordanjones243 opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #14 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

<!-- Generated by sourcery-ai[bot]: start summary -->

## Summary by Sourcery

Move theming support from beta to main by integrating design tokens for customizable styling, refactor component to use versioned Auro components, and update documentation to guide users on theming capabilities.

New Features:
- Introduce theming support for components, allowing customization through design tokens.

Enhancements:
- Refactor the component to use versioned Auro components for buttons and icons, improving maintainability and consistency.
- Update the component to use static HTML imports for better performance and modularity.

Documentation:
- Add documentation on how to restyle components using design tokens, providing examples for customization.

<!-- Generated by sourcery-ai[bot]: end summary -->

jordanjones243 committed:
feat: refactor color token structure with tier 3 tokens #14

jordanjones243 committed:
build: sync package-lock.json

jordanjones243 committed:
build(linter): use new eslint config

jordanjones243 committed:
feat(version): use custom versioned auro components

semantic-release-bot committed:
chore(release): 1.1.0-beta.1 [skip ci]

# [1.1.0-beta.1](https://github.com/AlaskaAirlines/auro-drawer/compare/v1.0.9...v1.1.0-beta.1) (2024-07-04)

### Features

* refactor color token structure with tier 3 tokens [#14](https://github.com/AlaskaAirlines/auro-drawer/issues/14) ([3c78a06](https://github.com/AlaskaAirlines/auro-drawer/commit/3c78a060203054b59f917635da8bd53aebd9b035))
* **version:** use custom versioned auro components ([8f8990c](https://github.com/AlaskaAirlines/auro-drawer/commit/8f8990c8c387b94775e06ebb67bbe418a360ead2))

jordanjones243 committed:
perf(button): use new beta version of flat variant button

semantic-release-bot committed:
chore(release): 1.1.0-beta.2 [skip ci]

# [1.1.0-beta.2](https://github.com/AlaskaAirlines/auro-drawer/compare/v1.1.0-beta.1...v1.1.0-beta.2) (2024-07-08)

### Performance Improvements

* **button:** use new beta version of flat variant button ([d6c9996](https://github.com/AlaskaAirlines/auro-drawer/commit/d6c99969fb8c12b67c324341b4da58e98caf54df))

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AlaskaAirlines/auro-drawer?pullRequest=22) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/AlaskaAirlines/auro-drawer?pullRequest=22) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: jordanjones243<br/>:x: semantic-release-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AlaskaAirlines/auro-drawer?pullRequest=22) it.</sub>

jordanjones243 committed:
fix: resolve minor color theming issues #14

BREAKING CHANGE: trigger major release for color theme support

jordanjones243 committed:
perf: update dependencies

jordanjones243 committed:
refactor: use auro-library build scripts

jordanjones243 committed:
feat: add tag name as attribute when custom registered

jordanjones243 committed:
perf: refactor custom component registration config

jordanjones243 committed:
docs: remove beta releases from CHANGELOG

semantic-release-bot committed:
chore(release): 2.0.0-beta.1 [skip ci]

# [2.0.0-beta.1](https://github.com/AlaskaAirlines/auro-drawer/compare/v1.1.0-beta.2...v2.0.0-beta.1) (2024-10-04)

### Bug Fixes

* resolve minor color theming issues [#14](https://github.com/AlaskaAirlines/auro-drawer/issues/14) ([c796a18](https://github.com/AlaskaAirlines/auro-drawer/commit/c796a18d78f22c71ebc1cf11c415ea582c979601))

### Features

* add tag name as attribute when custom registered ([8164fc4](https://github.com/AlaskaAirlines/auro-drawer/commit/8164fc47904b2779ffa049265f7b4efb994735d4))

### Performance Improvements

* refactor custom component registration config ([64cd8c3](https://github.com/AlaskaAirlines/auro-drawer/commit/64cd8c3b05e279f6690d5ce7f183dc4f15cc1405))
* update dependencies ([f3ef92e](https://github.com/AlaskaAirlines/auro-drawer/commit/f3ef92e95746da409c821bd7006a47ded2d975d4))

### BREAKING CHANGES

* trigger major release for color theme support

jordanjones243 commented:
:tada: This PR is included in version 2.0.0-beta.1 :tada:

The release is available on:
- [npm package (@beta dist-tag)](https://www.npmjs.com/package/@aurodesignsystem/auro-drawer/v/2.0.0-beta.1)
- [GitHub release](https://github.com/AlaskaAirlines/auro-drawer/releases/tag/v2.0.0-beta.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

jordanjones243 committed:
docs: remove 2.0.0 beta release from changelog

jordanjones243 commented:
:tada: This PR is included in version 2.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@aurodesignsystem/auro-drawer/v/2.0.0)
- [GitHub release](https://github.com/AlaskaAirlines/auro-drawer/releases/tag/v2.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
